controller/volume_controller.go

```diff
@@ -1358,24 +1307,32 @@ func getRebuildingReplicaCount(e *longhorn.Engine) int {
 func (vc *VolumeController) getReplenishReplicasCount(v *longhorn.Volume, rs map[string]*longhorn.Replica) int {
 	usableCount := 0
 	for _, r := range rs {
+		// The failed to schedule local replica shouldn't be counted
+		if !isDataLocalityDisabled(v) && r.Spec.HealthyAt == "" && r.Spec.FailedAt == "" && r.Spec.NodeID == "" &&
+			v.Status.CurrentNodeID != "" && r.Spec.HardNodeAffinity == v.Status.CurrentNodeID {
+			continue
+		}
 		// Skip the replica has been requested eviction.
 		if r.Spec.FailedAt == "" && (!r.Status.EvictionRequested) {
 			usableCount++
 		}
 	}
 
+	if v.Spec.NumberOfReplicas < usableCount {
```

Comment:
we need this check because when user decrease NumberOfReplicas, this function might return a negative number. With the combination with the check https://github.com/longhorn/longhorn-manager/pull/678/files#r483810861, the Longhorn will create infinite number of replicas.